src/Microsoft.AspNetCore.Routing.Abstractions/RouteValueDictionary.cs

```diff
@@ -513,7 +513,7 @@ public Enumerator(RouteValueDictionary dictionary)
                 _dictionary = dictionary;
```

Comment:
Can _dictionary be readonly?

Reply:
I'll have to retest it. This is mutable today to avoid issues with accessing readonly fields of structs. So reading through https://blogs.msdn.microsoft.com/seteplia/2018/05/03/avoiding-struct-and-readonly-reference-performance-pitfalls-with-errorprone-net/ there's probably no issue.

Reply:
I'm pretty sure the issue is readonly fields where the field type is a struct, not readonly fields on structs.

Reply:
I fixed this btw.